clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -286,4 +291,12 @@ private static FetchResponseData toMessage(Errors error,
                 .setSessionId(sessionId)
                 .setResponses(topicResponseList);
     }
+
+    private static FetchResponseData convertNullRecordsToEmpty(FetchResponseData fetchResponseData) {
+        fetchResponseData.responses().stream()
+            .flatMap(response -> response.partitions().stream())
+            .filter(partition -> partition.records() == null)
+            .forEach(partition -> partition.setRecords(MemoryRecords.EMPTY));
```

Comment:
The fetch path is perf sensitive and we are adding runtime cost every time even though this should never be required (i.e. the calling code should be doing the right thing already). Would a nested foreach loop be more efficient here?

Reply:
Indeed.
For a perf-sensitive fetch path, the stream operations are adding unnecessary stream creation overhead and lambda invocation costs. 🙇🏼